src/dodal/devices/i19/access_controlled/attenuator_motor_squad.py

```diff
@@ -0,0 +1,61 @@
+from typing import Annotated, Any, Self
+
+from ophyd_async.core import AsyncStatus
+from pydantic import BaseModel, model_validator
+from pydantic.types import PositiveInt, StringConstraints
+
+from dodal.devices.i19.access_controlled.blueapi_device import (
+    OpticsBlueAPIDevice,
+)
+from dodal.devices.i19.access_controlled.hutch_access import ACCESS_DEVICE_NAME
+
+PermittedKeyStr = Annotated[str, StringConstraints(pattern="^[A-Za-z0-9-_]*$")]
+
+
+class AttenuatorMotorPositionDemands(BaseModel):
+    continuous_demands: dict[PermittedKeyStr, float] = {}
+    indexed_demands: dict[PermittedKeyStr, PositiveInt] = {}
+
+    @model_validator(mode="after")
+    def no_keys_clash(self) -> Self:
+        common_keys = set(self.continuous_demands).intersection(self.indexed_demands)
+        common_key_count = sum(1 for _ in common_keys)
+        if common_key_count < 1:
+            return self
+        else:
+            ks: str = "key" if common_key_count == 1 else "keys"
+            error_msg = f"Common {ks} found in distinct motor demands: {common_keys}"
+            raise ValueError(error_msg)
+
+    def validated_complete_demand(self) -> dict[PermittedKeyStr, Any]:
+        return self.continuous_demands | self.indexed_demands
+
+
+class AttenuatorMotorSquad(OpticsBlueAPIDevice):
```

Comment:
Must: this device still needs to be instantiated in i19_1 and i19_2 so that the respective blueapi instances can pick it up